docs/benchmarking-workflow-corefx.md

```diff
@@ -0,0 +1,253 @@
+
+Every time you want to run the benchmarks against local build of CoreFX you need to provide the path to CoreRun:
+
+```cmd
+dotnet run -f netcoreapp3.0 --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" --filter $someFilter 
+```
+
+**Note:** BenchmarkDotNet expects a path to `CoreRun.exe` file (`corerun` on Unix), not to `Core_Root` folder.
+
+Once you rebuild the part of CoreFX you are working on, the appropriate `.dll` gets updated and the next time you run the benchmarks CoreRun is going to load the updated library.
+
+```cmd
+C:\Projects\corefx\src\System.Text.RegularExpressions\src> dotnet msbuild /p:ConfigurationGroup=Release
+```
+
+## Preventing Regressions
+
+Preventing regressions is one of the fundamental concepts of our performance culture.
+
+**Before introducing any changes**, you should run the benchmarks that test the performance of the feature that you are going to work on and store the results in a **dedicated** folder.
+
+```cmd
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0 \
+    --artifacts "C:\results\before" \
+    --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" \
+    --filter System.IO.Pipes*
+```
+
+Please try to **avoid running any resource-heavy processes** that could **spoil** the benchmark results while running the benchmarks.
+
+You can also create a **copy** of the folder with CoreRun and all the libraries to be able to run the benchmarks against the **unmodified base** in the future.
+
+After you introduce the changes and rebuild the part of CoreFX that you are working on **in Release** you should re-run the benchmarks. Remember to store the results in a different folder.
+
+```cmd
+C:\Projects\corefx\src\System.IO.Pipes\src> dotnet msbuild /p:ConfigurationGroup=Release
+
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0 \
+    --artifacts "C:\results\after" \
+    --coreRun "C:\Projects\corefx\artifacts\bin\runtime\netcoreapp-Windows_NT-Release-x64\CoreRun.exe" \
+    --filter System.IO.Pipes*
+```
+
+When you have the results you should use [ResultsComparer](../src/tools/ResultsComparer/README.md) to find out how your changes have affected the performance:
```

Comment:
Is there a way to compare your change (i.e. the "after") vs. what's in master (or some release branch) without building the "before" locally?

Reply:
very good question! I added an explanation about how to run against the latest .NET Core SDK:

https://github.com/adamsitnik/performance/blob/corefxDocs/docs/benchmarking-workflow-corefx.md#running-against-the-latest-net-core-sdk